engine/src/test/java/org/camunda/bpm/engine/impl/persistence/entity/util/TypedValueFieldTest.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.camunda.bpm.engine.impl.persistence.entity.util;
+
+import org.apache.commons.lang3.SerializationUtils;
+import org.camunda.bpm.engine.ProcessEngine;
+import org.camunda.bpm.engine.ProcessEngineConfiguration;
+import org.camunda.bpm.engine.impl.cfg.ProcessEngineConfigurationImpl;
+import org.camunda.bpm.engine.impl.interceptor.CommandExecutor;
+import org.camunda.bpm.engine.impl.variable.serializer.JavaObjectSerializer;
+import org.camunda.bpm.engine.impl.variable.serializer.TypedValueSerializer;
+import org.camunda.bpm.engine.impl.variable.serializer.ValueFieldsImpl;
+import org.camunda.bpm.engine.impl.variable.serializer.VariableSerializerFactory;
+import org.camunda.bpm.engine.variable.Variables;
+import org.camunda.bpm.engine.variable.value.TypedValue;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TypedValueFieldTest {
```

Comment:
I'd like to suggest writing tests that use only the API. That makes it more robust against future refactorings (and also us more certain that we are actually fixing the behavior from the user perspective).

You can test the case with disabled implicit updates as follows: Implement a serializable POJO class and implement `writeObject` (see 3.3 here: https://www.baeldung.com/java-serialization) that, depending on the value of a static flag, throws an exception. Then in the test, disable the flag when you create the variable initially and enable it before you retrieve it.

Lastly, you can move the test next to `org.camunda.bpm.engine.test.api.variables.ImplicitVariableUpdateTest`, e.g. call it `ImplicitVariableUpdateDisabledTest`.

Reply:
This looks good. Will update the PR with the suggested changes sometime this week.

Reply:
That's great, thank you.